aiidalab_widgets_base/viewers.py

```diff
@@ -546,9 +546,10 @@ def apply_selection(self, _=None):
 
     def download(self, change=None):  # pylint: disable=unused-argument
         """Prepare a structure for downloading."""
+        pk = f"pk-{self.pk}" if self.pk else "not-stored"
         self._download(
             payload=self._prepare_payload(),
-            filename="structure." + self.file_format.value,
+            filename=f"structure-{pk}.{self.file_format.value}",
```

Comment:
I don't think the "not-stored" suffix is needed:
```suggestion
            suffix = f"-{self.pk}" if self.pk else ""
            filename=f"structure{suffix}.{self.file_format.value}",
```
More importantly, I do not think we need the `self.pk` trait either:
```suggestion
            pk = getattr(self.structure, "pk", None)
            suffix = f"-{pk}" if pk is not None else ""
            filename=f"structure{suffix}.{self.file_format.value}",
```

Reply:
This wouldn't work, as `self.structure` is always an ASE object.

Reply:
Got it. Can you accept the first suggestion though?

Reply:
> Got it. Can you accept the first suggestion though?

Yes, but can you add back "pk" to the file name? This was a suggestion from @cpignedoli.

As far as I am aware, the term PK is widely used, so people are aware of what it means.

We should only make sure that people understand that the number in the file name is indeed the PK of the structure.

Reply:
Ah, yes, of course:
```suggestion
            suffix = f"-pk-{self.pk}" if self.pk else ""
            filename=f"structure{suffix}.{self.file_format.value}",
```